Parking_Lot_Reservation/Controllers/ParkingSpaceController.cs

```diff
@@ -0,0 +1,70 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using Parking_Lot_Reservation.Data;
+using Parking_Lot_Reservation.DTOs;
+using Parking_Lot_Reservation.Models;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Parking_Lot_Reservation.Controllers
+{
+    [ApiController]
+    [Route("api/[controller]")]
+    public class ParkingSpaceController : Controller
+    {
+        private readonly ApplicationDbContext _dbContext;
+
+        public ParkingSpaceController(ApplicationDbContext dbContext)
+        {
+            _dbContext = dbContext;
+        }
+
+        [HttpPost("addParkingSpace")]
+        public async Task<ActionResult<ParkingSpaceDTO>> Add([FromBody] ParkingSpaceDTO ParkingSpaceDTO)
```

Comment:
ParkingSpaceDTO.
Parameter names = camelCase

Reply:
Also I think returning ParkingSpaceModel isntead of ParkingSpaceDTO would be better as you'll instantly acknowledge Id of space you created

Reply:
Would't that destroy the purpose of DTOs? I don't quite understand why should i use model instead of DTO.

Reply:
At the moment youre basically responding with request. Cause as input for this method there is ParkingSpaceDTO and as response there is also ParkingSpaceDTO which when you think about it has no sense.
Having anything else than ParkingSpaceDTO would be better, it can be even Ok (http 200 response ) but then you have to change ActionResult<Parking...> to IActionResult.

DTO is used to limit data transferred between client and server, also excluding sensitive data.
It can be ez explained on Registration:
When you're requesting registration what you want to pass is only Username and Password.
Your app will create user in database and hash his password. To hash password usually you will get at least two or three things depending on used encription - PasswordHash and PasswordSalt.
Lets say all went good and user was created. Responding with PasswordHash and PasswordSalt would be quite stupid so you either respond with OK or respond with UserDTO which contains only his UserName and fixed text "Created"